app/services/district_report_service.rb

```diff
@@ -69,41 +98,40 @@ def registration_counts
-  # the previous three quarters. Each quarter cohort is made up of patients registered
-  # in the previous quarter who has had a follow up visit in the current quarter.
-  def compile_cohort_trend_data
-    Quarter.new(date: selected_date).downto(3).each do |results_quarter|
-      cohort_quarter = results_quarter.previous_quarter
+  def controlled_patients_count(time)
+    ControlledPatientsQuery.call(facilities: facilities, time: time).count
+  end
 
-      period = {cohort_period: :quarter,
-                registration_quarter: cohort_quarter.number,
-                registration_year: cohort_quarter.year}
-      query = MyFacilities::BloodPressureControlQuery.new(facilities: @facilities, cohort_period: period)
-      @data[:quarterly_registrations] << {
-        results_in: format_quarter(results_quarter),
-        patients_registered: format_quarter(cohort_quarter),
-        registered: query.cohort_registrations.count,
-        controlled: query.cohort_controlled_bps.count,
-        no_bp: query.cohort_missed_visits_count,
-        uncontrolled: query.cohort_uncontrolled_bps.count
-      }.with_indifferent_access
-    end
+  def percentage(numerator, denominator)
+    return 0 if denominator == 0
+    (numerator.to_f / denominator) * 100
+  end
+
+  def top_district_quarter_stats(district)
+    cohort_quarter = Quarter.current.previous_quarter
+    period = {cohort_period: :quarter,
+              registration_quarter: cohort_quarter.number,
+              registration_year: cohort_quarter.year}
+    query = MyFacilities::BloodPressureControlQuery.new(facilities: district.facilities, cohort_period: period)
+    controlled_count = query.cohort_controlled_bps.count
+    registrations_count = query.cohort_registrations.count
+    missed_visits_count = query.cohort_missed_visits_count
+    {
+      quarter_controlled_percentage: percentage(controlled_count, registrations_count),
+      quarter_missed_visits_percentage: percentage(missed_visits_count, registrations_count)
+    }
   end
 
-  def controlled_patients(time)
-    end_range = time.end_of_month
-    mid_range = time.advance(months: -1).end_of_month
-    beg_range = time.advance(months: -2).end_of_month
-    sub_query = LatestBloodPressuresPerPatientPerMonth
-      .select("distinct on (patient_id) *")
-      .under_control
-      .order("patient_id, bp_recorded_at DESC, bp_id")
-      .where(registration_facility_id: facilities)
-      .where("(year = ? AND month = ?) OR (year = ? AND month = ?) OR (year = ? AND month = ?)",
-        beg_range.year.to_s, beg_range.month.to_s,
-        mid_range.year.to_s, mid_range.month.to_s,
-        end_range.year.to_s, end_range.month.to_s)
-    LatestBloodPressuresPerPatientPerMonth.from(sub_query, "latest_blood_pressures_per_patient_per_months")
+  def top_district_benchmarks
+    districts_by_rate = FacilityGroup.all.each_with_object({}) { |district, hsh|
```

Comment:
TODO for future PR: figure out what orgs the current user has access to, and pass that into the service.